### PR TITLE
Auto-refresh on window activation

### PR DESCRIPTION
**Pull request by stefanhaller**
- **PR Description**

When you are making changes in your text editor or IDE, and then switch to the terminal window where lazygit is running, it doesn't immediately show you the latest changes you just made. You have to either refresh lazygit manually by typing shift-R (remembering that this command is only available in some panels but not others), or wait up to 10 seconds until the next auto-refresh timer comes along.

This PR solves this problem by automatically refreshing the files panel whenever lazygit's terminal window is brought to the front.

I have been running with this change locally for months now, and it has become indispensable for me; I always realize this whenever I have to work on a master-based branch that doesn't contain it.

Marked as a draft because it needs a change to gocui.

Note that an alternative solution to this problem might have been to subscribe to filesystem notifications and refresh whenever we are notified that files have changed. It seems that lazygit has experimented with this in the past, there are some rudiments of it in the code, but it's disabled. However, I actually think that refreshing on window activation is superior: with filesystem notifications you have to be careful not to refresh too often. Many files can be changed at once, and this probably needs some timer-based heuristic to wait until activity has died down before you refresh. Doing it on window activation doesn't have this problem.

Of all the other git clients I know, there's only one that reacts to filesystem changes (Sublime Merge), all the others refresh on window activation.

- **Please check if the PR fulfills these requirements**

* [x] Cheatsheets are up-to-date (run `go run scripts/cheatsheet/main.go generate`)
* [x] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [ ] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
* [ ] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [ ] Docs (specifically `docs/Config.md`) have been updated if necessary
* [x] You've read through your own file changes for silly mistakes etc

### CODE REVIEW

**Comment by jesseduffield** [comment]
Holy shit this is a gamechanger

**Comment by jesseduffield** [comment]
Is there a PR against tcell for this?

**Comment by stefanhaller** [comment]
The tcell PR is mentioned in the commit message of the first commit, it's https://github.com/gdamore/tcell/pull/599 (I see that you found it already meanwhile 😄). For some reason I didn't get any further reactions on it from Garrett since March; I hope he's doing well.

I also have to say that the PR can't be merged in it's current state because there's a change missing for the `wScreen` class (which I only realized very recently), and I don't know how to do that. Help would be appreciated. However, for our purposes it's good enough, since we don't use that class, so switching to my fork for now seems ok to me.

**Comment by jesseduffield** [comment]
Okay that sounds fine to me. I'm happy with this PR as-is (I see it's in draft: what else did you intend to change?)

File: vendor/github.com/jesseduffield/gocui/gui.go [hunk 1]
```diff
@@ -726,6 +731,7 @@ func (g *Gui) MainLoop() error {
 
 	if g.Mouse {
 		Screen.EnableMouse()
+		Screen.EnableFocus()
```
[the remark this answers is not in the document]
Comment (jesseduffield) [reply vendor/github.com/jesseduffield/gocui/gui.go:734]: But does it still require `Mouse: true`? Strange that it's a mouse event if it doesn't necessarily involve a mouse click

**Comment by jesseduffield** [comment]
I've added you as a collaborator in gocui so if you want you can get these changes into gocui yourself:

I often make changes in lazygit's vendor directory then copy them across to my gocui repo with
cp -r /path/to/lazygit/vendor/github.com/jesseduffield/gocui /path/to/gocui

Then I get that merged to gocui and then back in the lazygit repo I run:

scripts/bump_gocui.sh

File: vendor/github.com/jesseduffield/gocui/gui.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jesseduffield) [reply vendor/github.com/jesseduffield/gocui/gui.go:734]: I don't think it needs its own flag though. The purpose of having the Mouse flag (afaik) is that it's easier to drag and select text when mouse mode is off (in iterm I need to hold option to do it)

[the remark this answers is not in the document]
Comment (stefanhaller) [reply vendor/github.com/jesseduffield/gocui/gui.go:734]: Ah ok. Just enable focus unconditionally then?

[the remark this answers is not in the document]
Comment (jesseduffield) [reply vendor/github.com/jesseduffield/gocui/gui.go:734]: Yep, we'll see if it causes issues for anybody before we make it configurable

**Comment by stefanhaller** [comment]
@jesseduffield This is ready to go if you want to give it a final look. It hasn't changed from the draft except for the middle commit. Feel free to merge it if you're happy with it, I'm not sure I would get around to that today.

**Comment by jesseduffield** [comment]
Super excited to use this myself. I'll clean up the file watcher code now that we've got this in place

**Pull request merged**